plugin/parser/differ/mysql/differ.go

```diff
@@ -312,6 +371,72 @@ func buildTableMap(nodes []ast.StmtNode) map[string]*ast.CreateTableStmt {
 	return tableMap
 }
 
+// buildViewMap returns a map of view name to create view statements.
+func buildViewMap(nodes []ast.StmtNode) map[string]*ast.CreateViewStmt {
+	viewMap := make(map[string]*ast.CreateViewStmt)
+	for _, node := range nodes {
+		if stmt, ok := node.(*ast.CreateViewStmt); ok {
+			viewName := stmt.ViewName.Name.O
+			viewMap[viewName] = stmt
+		}
+	}
+	return viewMap
+}
+
+// getTempView returns the temporary view name and the create statement.
+func getTempView(stmt *ast.CreateViewStmt) *ast.CreateViewStmt {
+	// We create the temp view similar to what mysqldump does.
+	// Create a temporary view with the same name as the view. Its columns should
+	// have the same name in order to satisfy views that depend on this view.
+	// This temporary view will be removed when the actual view is created.
+	// The column properties are unnecessary and not preserved in this temporary view.
+	// because other views only need to reference the column name.
+	//  Example: SELECT 1 AS colName1, 1 AS colName2.
+	// TODO(zp): support SDL for GitOps.
+	var selectFileds []*ast.SelectField
+	// mysqldump always show field list
+	if len(stmt.Cols) > 0 {
+		for _, col := range stmt.Cols {
+			selectFileds = append(selectFileds, &ast.SelectField{
+				Expr: &driver.ValueExpr{
+					Datum: types.NewDatum(1),
+				},
+				AsName: col,
+			})
+		}
+	} else {
+		for _, field := range stmt.Select.(*ast.SelectStmt).Fields.Fields {
+			var fieldName string
+			if field.AsName.O != "" {
+				fieldName = field.AsName.O
+			} else {
+				fieldName = field.Expr.(*ast.ColumnNameExpr).Name.Name.O
+			}
+			selectFileds = append(selectFileds, &ast.SelectField{
+				Expr: &driver.ValueExpr{
+					Datum: types.NewDatum(1),
+				},
+				AsName: model.NewCIStr(fieldName),
+			})
+		}
+	}
+
+	return &ast.CreateViewStmt{
+		ViewName: stmt.ViewName,
+		Select: &ast.SelectStmt{
+			SelectStmtOpts: &ast.SelectStmtOpts{
+				SQLCache: true,
+			},
+			Fields: &ast.FieldList{
+				Fields: selectFileds,
+			},
+		},
+		OrReplace: true,
+		Definer:   stmt.Definer,
```

Comment:
Do we also need these definer and security?

Reply:
To avoid generating panic.https://sourcegraph.com/github.com/pingcap/tidb/-/blob/parser/ast/ddl.go?L1398